Shen rework + removal of Hakutaku's triple attack mod + remove drop pools from mob bst's pets

dallano opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?
Created new branch to organize NM_Mob_Fixes into a single commit as well as shaved off irrelevant and or unnecessary changes.

Changes included:
Shen rework
Removal of Hakutaku's triple attack mod upon request.
Removal of mob beastmaster pet's ability to drop items.

## Steps to test these changes

<!-- Clear and detailed steps to test your changes here -->

dallano committed:
Rebasing NM_Mob_Fixes

Created new branch to organize NM_Mob_Fixes into a single commit as well as shaved off irrelevant changes.

Changes included:
Shen rework
Removal of Hakutaku's triple attack mod upon request.
Removal of mob beastmaster pet's ability to drop items.

claywar commented on scripts/zones/Bibiki_Bay/mobs/Shen.lua:
```diff
@@ -2,14 +2,23 @@
 entity.onMobFight = function(mob, target)
-    if (mob:getBattleTime() % 45 == 0) then
-        local mobId = mob:getID()
+    entity.spawnFiltrate(mob, target)
```
This can just be a local function, and does not need to be attached to the entity table

dallano committed:
Fixes

Fixed spawning of the filtrate's to be handled locally
Added spell lists for Shen and filtrates.

dallano committed:
Update Shen.lua

Quick fix.

dallano committed:
Update Shen.lua

Another embarrassing fix in the changing of a local function.

dallano committed:
Update mob_spell_lists.sql

SQL duplicate name fix.

dallano committed:
Update mob_spell_lists.sql

Missing fix

Mortalelite commented on scripts/zones/Bibiki_Bay/mobs/Shen.lua:
```diff
@@ -2,14 +2,23 @@
 --  Mob: Shen
 -----------------------------------
+local ID = require("scripts/zones/Bibiki_Bay/IDs")
+require("scripts/globals/mixins/families/uragnite")
```
we would point to 

`scripts/mixins/families/uragnite`

dallano committed:
Update Shen.lua

Mortalelite commented:
Looks good to go, just need a review.

zach2good reviewed (approved):
Seems fine, squash on merge

(Please consider keeping the title of your PR, and your commits, somewhat clean looking. Once they become part of the history it's next to impossible for us to change it👍)

claywar commented on scripts/zones/Bibiki_Bay/mobs/Shen.lua:
```diff
@@ -2,36 +2,29 @@
-        for i = 1, 2 do
-            if (not GetMobByID(mobId+i):isSpawned()) then
-                SpawnMob(mobId+i):updateEnmity(target)
+    local spawnFiltrate = function()
```
Is there a reason for defining a function here only to call it right below?  Seems redundant

dallano replied:
I was told to make this a local function. I must have been confused lol. Should I keep it local and just move it outside of onMobFight?

claywar replied:
You're right, and I apologise for my original review.  I meant more that it shouldn't be attached to the object itself, since its unnecessary.  In this case, the function is only used in one spot, so there's really not a need to define a function.

```
    local spawnFiltrate = function()
        if mob:getAnimationSub() == 1 and mob:getHP() > 0 then
            for i = 1, 2 do
                if not GetMobByID(mob:getID()+i):isSpawned() then
                    SpawnMob(mob:getID()+i):updateEnmity(target)
                end
            end
        end
    end

    spawnFiltrate()
```
That's the same as saying:
```
      if mob:getAnimationSub() == 1 and mob:getHP() > 0 then
          for i = 1, 2 do
              if not GetMobByID(mob:getID()+i):isSpawned() then
                  SpawnMob(mob:getID()+i):updateEnmity(target)
              end
          end
      end
```

dallano replied:
Oh! Most definitely. I'll try and employ this more in the future. Thank you

dallano committed:
Update Shen.lua

Making function local to remove redundancy

TeoTwawki commented:
Your one of your previous commits (diff pr? comments appear to have been lost..) used different values than the mixin for Shen's shell mods and I had suggested we (not necessarily you, if you aren't comfortable doing it) modify the mixin to facilitate this. I just want to know if shen needs different values or if using the mixin as is has turned out to be retail-accurate. That is all.

dallano commented:
I can definitely hop on this! you guys at LSB and over at ASB have been extremely supportive with teaching me how things run and what not. I'll make sure to push some proper tables and mixins soon. Or at least report some findings.
